src/transformStateWithClones.js

```diff
@@ -7,7 +8,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const modifiedState = { ...state };
+  const modifiedActions = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(modifiedState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete modifiedState[key];
+        }
+        break;
+
+      case 'clear':
+        for (const property in modifiedState) {
+          delete modifiedState[property];
+        }
```

Comment:
Here you can use one line: modifiedState = {}, without using a loop )

Reply:
Ok, will try your solution

Reply:
Yeah, without loop it looks much more better;)